src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -85,3 +99,61 @@ export function euiNavLink(
     'data-test-subj': 'navDrawerAppsMenuLink',
   };
 }
+
+// Providing a buffer between the limit and the cut off index
+// protects from truncating just the last couple (6) characters
+const TRUNCATE_LIMIT: number = 64;
+const TRUNCATE_AT: number = 58;
+
+function truncateRecentItemLabel(label: string): string {
+  if (label.length > TRUNCATE_LIMIT) {
+    label = `${label.substring(0, TRUNCATE_AT)}…`;
+  }
+
+  return label;
+}
+
+/**
+ * @param {string} url - a relative or root relative url.  If a relative path is given then the
+ * absolute url returned will depend on the current page where this function is called from. For example
+ * if you are on page "http://www.mysite.com/shopping/kids" and you pass this function "adults", you would get
+ * back "http://www.mysite.com/shopping/adults".  If you passed this function a root relative path, or one that
+ * starts with a "/", for example "/account/cart", you would get back "http://www.mysite.com/account/cart".
+ * @return {string} the relative url transformed into an absolute url
+ */
+function relativeToAbsolute(url: string) {
+  const a = document.createElement('a');
+  a.setAttribute('href', url);
+  return a.href;
+}
+
+export type RecentNavLink = ReturnType<typeof createRecentNavLink>;
+
+export function createRecentNavLink(
+  recentLink: ChromeRecentlyAccessedHistoryItem,
+  navLinks: ChromeNavLink[],
+  basePath: HttpStart['basePath']
+) {
+  const { link, label } = recentLink;
+  const href = relativeToAbsolute(basePath.prepend(link));
+  const navLink = navLinks.find(nl => href.startsWith(nl.baseUrl ?? nl.subUrlBase));
+  let titleAndAriaLabel = label;
+
+  if (navLink) {
+    titleAndAriaLabel = i18n.translate('core.ui.recentLinks.linkItem.screenReaderLabel', {
+      defaultMessage: '{recentlyAccessedItemLinklabel}, type: {pageType}',
+      values: {
+        recentlyAccessedItemLinklabel: label,
+        pageType: navLink.title,
+      },
+    });
+  }
+
+  return {
+    href,
+    label: truncateRecentItemLabel(label),
+    title: titleAndAriaLabel,
+    'aria-label': titleAndAriaLabel,
+    iconType: navLink?.euiIconType,
```

Comment:
```suggestion
```

I'd remove this. We're moving away from using the icons for the individual apps, and since this is only available sometimes it makes for a busy view. Here's what it looks like without the icon.

![image](https://user-images.githubusercontent.com/324519/80779257-f80bee80-8b1f-11ea-9fea-8332b3f4717b.png)

Reply:
@myasonik Wanna make sure this change doesn't get lost. Should be a simple commit (gave you the code above). Might need to update your snaps after.

Reply:
Did this just for the new nav here: https://github.com/elastic/kibana/pull/64018/files?file-filters%5B%5D=.js&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx#diff-daa5d177f3f019311051969fd0a63ab1R154-R159